### ISSUE TITLE
Native support for Windows

### ISSUE THREAD

**Issue by Halbaroth** [OCamlPro/alt-ergo#919]
After #584, we stop building and testing Alt-Ergo on Windows with Cygwin. It seems that we have users on Windows. We should build and test AE on Windows again.

**Comment by Halbaroth** [comment]
Besides, I hope `Dolmen` works on Windows. I think @Gbury  did the job to make it works ;)

**Comment by Gbury** [comment]
Yes, `dolmen` works on windows ! I've worked hard to ensure that (and I have CI to check for that as well).

**Comment by Gbury** [comment]
Small caveat: on windows, time limits are slightly less reliable since they use the GC alarm mechanism instead of Unix timers, but I don't think it's a problem.

**Comment by bclement-ocp** [comment]
IIRC the problem before #584 was not that Dolmen did not work for Windows but that the version we used was not available on the Windows OPAM repository ( https://github.com/ocaml-opam/opam-repository-mingw ) which is lagging behind the "main" OPAM repository. Currently we use a pinned version so that shouldn't be a problem to reinstate the build; if/when we make a release, we will need to take care of this.

(Note: if my understanding of [this](https://github.com/ocaml/setup-ocaml/pull/651) is correct, `opam-repository-mingw` should no longer be necessary once OPAM 2.2 is release, which would make the problem go away — in fact I do not understand if it is currently required or not)

**Comment by bclement-ocp** [comment]
Additional note: interested users clarified that they wanted *native* Windows support, not Cygwin. AFAIK that means either waiting for OPAM 2.2 or using Diskuv. I'd say we wait for OPAM 2.2 as there is no pressing need.

**Comment by Halbaroth** [comment]
PR #1203 restores Windows CI with `ocaml-setup`. Unfortunately, `ocaml-setup` does not support `MSYS2`. I opened an issue 
https://github.com/ocaml/setup-ocaml/issues/846 and a contributor of `ocaml-setup` submitted a PR to fix it https://github.com/ocaml/setup-ocaml/pull/857. 

I suggest to test Alt-Ergo on Windows + MSYS2 before publishing the release and I remove the milestone on this issue.
